winit-wayland/src/window/state.rs

```diff
@@ -1164,6 +1168,10 @@ impl Drop for WindowState {
             viewport.destroy();
         }
 
+        if let Some(xdg_toplevel_icon_manager) = self.xdg_toplevel_icon_manager.take() {
+            xdg_toplevel_icon_manager.destroy();
+        }
+
```

Comment:
I think this one is actually wrong, since the manager is global and thus should be dropped when dropping the global state? Right now I think it'll get dropped and then the next window won't be able to use the `icon`.

Though, we generally drop managers, so just drop this.

Reply:
so should I add a drop implementation directly in `XdgToplevelIconManagerState`?

Reply:
oh, so i just remove these lines and everything should be good?